qcodes/instrument_drivers/QDevil/QDevil_QDAC.py

```diff
@@ -0,0 +1,1061 @@
+# QCoDeS driver for the QDevil QDAC using channels
+# Adapted by QDevil from "qdev\QDac_channels.py" in
+# the instrument drivers package
+# Version 2.1 QDevil 2020-02-10
+
+import time
+import visa
+import logging
+
+from functools import partial
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+from qcodes.instrument.channel import MultiChannelInstrumentParameter
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.utils import validators as vals
+from enum import Enum
+from collections import namedtuple
+
+LOG = logging.getLogger(__name__)
+
+
+_ModeTuple = namedtuple('Mode', 'v i')
+
+
+class Mode(Enum):
+    """
+    Enum type use as the mode parameter for channels
+    defining the combined voltage and current range.
+
+    get_label() returns a text representation of the mode.
+    """
+    vhigh_ihigh = _ModeTuple(v=0, i=1)
+    vhigh_ilow = _ModeTuple(v=0, i=0)
+    vlow_ilow = _ModeTuple(v=1, i=0)
+
+    def get_label(self):
+        _MODE_LABELS = {
+                self.vhigh_ihigh: "V range high / I range high",
+                self.vhigh_ilow: "V range high / I range low",
+                self.vlow_ilow: "V range low / I range low"}
+        return _MODE_LABELS[self]
+
+
+class Waveform:
+    # Enum-like class defining the built-in waveform types
+    sine = 1
+    square = 2
+    triangle = 3
+    staircase = 4
+    all_waveforms = [sine, square, triangle, staircase]
+
+
+class Generator():
+    #  Class used in the internal book keeping of generators
+    def __init__(self, generator_number):
+        self.fg = generator_number
+        self. t_end = 9.9e9
```

Comment:
Great that you had a chance to push all your updates, Anders! I am just having a quick browse through and I noticed this additional ' '-space at `self.t_end`. I will get back to you later.

Reply:
Ha! (you know what I mean)